lib/tests/streamlit/runtime/state/query_params_test.py

```diff
@@ -0,0 +1,139 @@
+            )
+
+    def test_getitem_list(self):
+        print(self._query_params)
+        # get the last item in the array
+        assert self._query_params["two"] == "y"
+
+    def test_getitem_single_element_int(self):
+        # Internal state is always a string so we test setting it here
+        self._query_params["baz"] = 1
+        assert self._query_params["baz"] == "1"
+
+    def test_getitem_single_element_float(self):
+        # Internal state is always a string so we test setting it here
+        self._query_params["corge"] = 1.23
+        assert self._query_params["corge"] == "1.23"
+
+    def test__getitem__value(self):
+        assert self._query_params["foo"] == "bar"
+
+    def test_get(self):
+        assert self._query_params.get("foo") == "bar"
+
+    def test_get_default(self):
+        return_val = self._query_params.get("dog", default="bark")
+        assert return_val == "bark"
+
+    def test__getattr__(self):
+        assert self._query_params.foo == "bar"
+
+    def test__setitem__query_params(self):
+        assert "test" not in self._query_params
+        self._query_params["test"] = "test"
+        assert self._query_params.get("test") == "test"
+        message = self.get_message_from_queue(0)
+        assert (message.page_info_changed.query_string, "test=test")
+
+    def test__setitem_empty_string(self):
+        assert "test" not in self._query_params
+        self._query_params["test"] = ""
+        assert self._query_params["test"] == ""
+        message = self.get_message_from_queue(0)
+        assert (message.page_info_changed.query_string, "test=")
+
+    def test__setattr__(self):
+        self._query_params.test = "test"
+        assert self._query_params.get("test") == "test"
+
+    def test__setattr__list(self):
+        self._query_params.test = ["test", "test2"]
+        print(self._query_params)
+        assert self._query_params["test"] == "test2"
+        message = self.get_message_from_queue(0)
+        assert (message.page_info_changed.query_string, "test=test&test=test2")
+
+    def test_getall_nonexistent(self):
+        assert self._query_params.get_all("nonexistent") == []
+
+    def test_getall_single_element(self):
+        assert self._query_params.get_all("foo") == ["bar"]
+
+    def test_getall_list(self):
+        assert self._query_params.get_all("two") == ["x", "y"]
+
+    def test_getall_list_non_str_and_empty_str(self):
+        self._query_params["test"] = ["", "a", 1, 1.23]
+        assert self._query_params.get_all("test") == ["", "a", "1", "1.23"]
+
+    def test_contains_valid(self):
+        assert "foo" in self._query_params
+
+    def test_contains_invalid(self):
+        assert "baz" not in self._query_params
+
+    def test_clear(self):
+        self._query_params.clear()
+        assert len(self._query_params) == 0
+        message = self.get_message_from_queue(0)
+        self.assertEqual(message.page_info_changed.query_string, "")
```

Comment:
We shouldn't be using ` self.assertEqual` here since we're already using plain `assert`s throughout this file (and are trying to do so in general because it's cleaner + `pytest` formats their output nicely